Netcracker/KubeMarine
# [CPDEV-95906] IPIP Encapsulation check

### Description
* Calico IPIP traffic encapsulation needs to be covered by `check_iaas`
* The check should work on running cluster and clean nodes as well

### Solution
* Add check for IP in IP overlay into `check_iaas` procedure
* Create additional code that implements IPIP packet sending and receiving
*  Change CI process to build additional binary
* Update Installation and Kubecheck guides


### Test Cases

**TestCase 1**
New check is working on clean nodes

Test Configuration:

- Hardware: 4CPU/8GB
- OS: Ubuntu 22.04; CentOS 7; RHEL 8; Rocky Linux 8; Oracle Linux 7
- Inventory: miniHA

Steps:

1. Create a set of clean nodes
2. Set `plugins.calico.mode=ipip` in `cluster.yaml`
3. Run `check_iaas` procedure with `--task='network.ipip_connectivity'`

Results:

| Before | After |
| ------ | ------ |
| Not Applicable | IP in IP is checked |

**TestCase 2**
Exemptions in `network.ipip_connectivity` check are working

Test Configuration:

- Hardware: 4CPU/8GB
- OS: Ubuntu 22.04
- Inventory: miniHA

Steps:

1. Set `plugins.calico.install=false` in `cluster.yaml`
2. Run `check_iass` procedure with `--task='network.ipip_connectivity'`
4. Set `plugins.calico.mode=vxlan` in `cluster.yaml`
5. Run `check_iass` procedure with `--task='network.ipip_connectivity'`
6. Set IPv6 network for `services.kubeadm.networking.podSubnet` in `cluster.yaml`
4. Run `check_iass` procedure with `--task='network.ipip_connectivity'`

Results:

| Before | After |
| ------ | ------ |
| Not Applicable | check is skipped due to `Calico` is not set as CNI |
| Not Applicable | check is skipped due to encapsulation is not set to `ipip` |
| Not Applicable | check is skipped due to IP in IP encapsulation supports only IPv4 addresses |

**TestCase 3**
The check is working on running cluster

Test Configuration:

- Hardware: 4CPU/8GB
- OS: Ubuntu 22.04
- Inventory: fullHA

Steps:

1.  Run `check_iass` procedure with `--task='network.ipip_connectivity'` on running cluster. Calico must be running in `ipip` mode

Results:

| Before | After |
| ------ | ------ |
| Not Applicable | Success |


### Checklist
